src/sources/libfaadloader.cpp

```diff
@@ -0,0 +1,190 @@
+            m_pLibrary->resolve("NeAACDecPostSeekReset"));
+    m_neAACDecDecode2 = reinterpret_cast<NeAACDecDecode2_t>(
+            m_pLibrary->resolve("NeAACDecDecode2"));
+    m_neAACDecGetErrorMessage = reinterpret_cast<NeAACDecGetErrorMessage_t>(
+            m_pLibrary->resolve("NeAACDecGetErrorMessage"));
+
+    if (    !m_neAACDecOpen ||
+            !m_neAACDecGetCurrentConfiguration ||
+            !m_neAACDecSetConfiguration ||
+            !m_neAACDecInit2 ||
+            !m_neAACDecClose ||
+            !m_neAACDecPostSeekReset ||
+            !m_neAACDecDecode2 ||
+            !m_neAACDecGetErrorMessage
+    ) {
+        kLogger.debug() << "NeAACDecOpen:" << m_neAACDecOpen;
+        kLogger.debug() << "NeAACDecGetCurrentConfiguration:" << m_neAACDecGetCurrentConfiguration;
+        kLogger.debug() << "NeAACDecSetConfiguration:" << m_neAACDecSetConfiguration;
+        kLogger.debug() << "NeAACDecInit2:" << m_neAACDecInit2;
+        kLogger.debug() << "NeAACDecClose:" << m_neAACDecClose;
+        kLogger.debug() << "NeAACDecPostSeekReset:" << m_neAACDecPostSeekReset;
+        kLogger.debug() << "NeAACDecDecode2:" << m_neAACDecDecode2;
+        kLogger.debug() << "NeAACDecGetErrorMessage:" << m_neAACDecGetErrorMessage;
+        m_neAACDecOpen = nullptr;
+        m_neAACDecGetCurrentConfiguration = nullptr;
+        m_neAACDecSetConfiguration = nullptr;
+        m_neAACDecInit2 = nullptr;
+        m_neAACDecClose = nullptr;
+        m_neAACDecPostSeekReset = nullptr;
+        m_neAACDecDecode2 = nullptr;
+        m_neAACDecGetErrorMessage = nullptr;
+        m_pLibrary->unload();
+        m_pLibrary.reset();
+        return;
+    }
+    kLogger.debug() << "Successfully loaded faad2 library " << m_pLibrary->fileName();
+};
+
+bool LibFaadLoader::isLoaded() {
+    if (m_pLibrary) {
+        return m_pLibrary->isLoaded();
+    }
+    return false;
+}
+
+LibFaadLoader::Handle LibFaadLoader::Open() {
+    if (m_neAACDecOpen) {
+        return m_neAACDecOpen();
+    }
+    return nullptr;
+}
+
+LibFaadLoader::Configuration*
+LibFaadLoader::GetCurrentConfiguration(Handle hDecoder) {
+    if (m_neAACDecGetCurrentConfiguration) {
+        return m_neAACDecGetCurrentConfiguration(hDecoder);
+    }
+    return nullptr;
+}
+
+unsigned char LibFaadLoader::SetConfiguration(
+        Handle hDecoder, Configuration* pConfig) {
+    if (m_neAACDecSetConfiguration) {
+        return m_neAACDecSetConfiguration(hDecoder, pConfig);
+    }
+    return 0;
+}
+
+// Init the library using a DecoderSpecificInfo
+char LibFaadLoader::Init2(
+        Handle hDecoder,
+        unsigned char* pBuffer,
+        unsigned long SizeOfDecoderSpecificInfo,
+        unsigned long* pSamplerate,
+        unsigned char* pChannels) {
+    if (m_neAACDecInit2) {
+        return m_neAACDecInit2(hDecoder,
+                pBuffer,
+                SizeOfDecoderSpecificInfo,
+                pSamplerate,
+                pChannels);
+    }
+    return 0;
```

Comment:
-1 would be more appropriate, because 0 is successful.

Reply:
No see: 
https://github.com/gypified/libfaad/blob/master/docs/Ahead%20AAC%20Decoder%20library%20documentation.pdf
> Return values:
> 0 – Error, invalid configuration.
> 1 – OK

I will add a comment.

Reply:
NeAACDecInit2
Return values:
< 0 – Error
0 - OK

Reply:
Ups, I have fixed this on the wrong function. Now the new comment is in both places.